static/js/components/channels/withChannelCreateDialog.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+/* global SETTINGS: false */
+import React from "react"
+import R from "ramda"
+
+import { showDialog, hideDialog, setToastMessage } from "../../actions/ui"
+import {
+  startChannelEdit,
+  clearChannelEdit,
+  updateChannelEdit,
+  createChannel
+} from "../../actions/channels"
+import { TOAST_SUCCESS } from "../../constants"
+import { actions } from "../../lib/redux_rest"
+import { discussionErrors } from "../../lib/validation/discussions"
+import { getDisplayName } from "../../util/util"
+import { CHANNEL_CREATE_DIALOG } from "../../constants"
+import ChannelCreateDialog from "./ChannelCreateDialog"
+
+import type { CreateChannelResponse } from "../../flow/discussionTypes"
+
+const isVisible = R.propOr(false, CHANNEL_CREATE_DIALOG)
+
+export const withChannelCreateDialog = (WrappedComponent: ReactClass<*>) => {
+  class WithChannelCreateDialog extends React.Component {
+    openChannelCreateDialog = (searchkit: Object): void => {
+      const { dispatch } = this.props
+      dispatch(
+        startChannelEdit({
+          filters: searchkit.query.getSelectedFilters(),
+          searchkit
+        })
+      )
+      dispatch(showDialog(CHANNEL_CREATE_DIALOG))
+    }
+
+    closeAndClearDialog = (): void => {
+      const { dispatch } = this.props
+      dispatch(clearChannelEdit())
+      dispatch(hideDialog(CHANNEL_CREATE_DIALOG))
+    }
+
+    closeAndCreateDialog = (): void => {
+      const { dispatch, channelDialog: { inputs, searchkit } } = this.props
```

Comment:
Can you write some tests for this function? It doesn't look like this is covered by code coverage